src/vs/workbench/browser/parts/editor/textEditor.ts

```diff
@@ -102,7 +102,7 @@ export abstract class AbstractTextEditor<T extends IEditorViewState> extends Abs
 	}
 
 	protected shouldHandleConfigurationChangeEvent(e: ITextResourceConfigurationChangeEvent, resource: URI | undefined): boolean {
-		return e.affectsConfiguration(resource, 'editor');
+		return e.affectsConfiguration(resource, 'editor') || e.affectsConfiguration(resource, 'problems');
```

Comment:
Not a big fan of this: the setting is from a `contrib`, while this editor is in core. This creates a soft dependency of core to `contrib`, but it should be the other way around. There needs to be a follow up debt item to clean this up if we cannot address this now.